### PR TITLE
go: update to 1.22.0

### PR DESCRIPTION
**Pull request by felix**
This also includes a new dependency of go1.22-bootstrap.

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (amd64-libc)

### CODE REVIEW

**Comment by felix** [comment]
I will have to find some time to look at the musl ports. See here https://github.com/golang/go/issues/59305. Or suggestions welcome from those with time or musl know-how.

**Comment by leahneukirchen** [comment]
Don't use Go 1.20 for bootstrapping, use 1.21 or 1.22, there the official binaries are statically linked and should run fine on musl.

**Comment by felix** [comment]
> Don't use Go 1.20 for bootstrapping, use 1.21 or 1.22, there the official binaries are statically linked and should run fine on musl.

Thanks, I didn't realise things had changed.

File: srcpkgs/go1.22-bootstrap/template [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		;;
+esac
+
+distfiles="https://dl.google.com/go/go${version}.linux-${_dist_arch}.tar.gz"
+
+do_install() {
+	vmkdir usr/lib/go${version}
+	vcopy bin usr/lib/go${version}
+	vcopy src usr/lib/go${version}
+	vcopy pkg usr/lib/go${version}
```
Comment (classabbyamp) [inline srcpkgs/go1.22-bootstrap/template:49]: does it make sense to make this `go1.22` (`go${version%.*}`) so the main template doesn't have to change if the bootstrap is changed?

Comment (leahneukirchen) [reply srcpkgs/go1.22-bootstrap/template:49]: Yes, but there's also no good reason to update the bootstrap I think (unless a serious miscompilation is detected).

**Pull request merged**